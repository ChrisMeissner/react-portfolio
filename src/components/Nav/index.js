import React from 'react';

const navContainerStyle = {
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'center',
  alignItems: 'center',
  width: '60%',
  height: 'inherit'
};

const navAnchorStyle = {
  fontSize: '40px',
  color: 'white',
  textDecoration: 'none',
  justifyContent: 'center',
  padding: '0% 2.5%'
}

function Nav() {
  return (
      <header style={navContainerStyle}>
        <a href="#about" style={navAnchorStyle}>
          <p>About Me</p>
        </a>
        <a href="#portfolio" style={navAnchorStyle}>
          <p>Portfolio</p>
        </a>
        <a href="#contact" style={navAnchorStyle}>
          <p>Contact</p>
        </a>
        <a href="#resume" style={navAnchorStyle}>
          <p>Resume</p>
        </a>
      </header>
  );
};

export default Nav;