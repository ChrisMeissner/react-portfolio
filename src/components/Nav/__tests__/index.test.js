import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

describe('Nav Component', () => {
  // renders Nav test
  // first test
  it('renders', () => {
    render(<Nav/>);
  });

  // second test
  it('matches snapshot DOM node structure', () => {
    // render Nav
    const { asFragment } = render(<Nav />);
    expect(asFragment()).toMatchSnapshot();
  })

});