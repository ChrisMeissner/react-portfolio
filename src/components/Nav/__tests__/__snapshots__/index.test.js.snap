// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <header
    style="display: flex; flex-direction: row; justify-content: center; align-items: center; width: 60%; height: inherit;"
  >
    <a
      href="#about"
      style="font-size: 40px; color: white; text-decoration: none; justify-content: center; padding: 0% 2.5%;"
    >
      <p>
        About Me
      </p>
    </a>
    <a
      href="#portfolio"
      style="font-size: 40px; color: white; text-decoration: none; justify-content: center; padding: 0% 2.5%;"
    >
      <p>
        Portfolio
      </p>
    </a>
    <a
      href="#contact"
      style="font-size: 40px; color: white; text-decoration: none; justify-content: center; padding: 0% 2.5%;"
    >
      <p>
        Contact
      </p>
    </a>
    <a
      href="#resume"
      style="font-size: 40px; color: white; text-decoration: none; justify-content: center; padding: 0% 2.5%;"
    >
      <p>
        Resume
      </p>
    </a>
  </header>
</DocumentFragment>
`;
